Pcap++/src/DpdkDeviceList.cpp

```diff
@@ -192,11 +183,15 @@ bool DpdkDeviceList::initDpdkDevices(uint32_t mBufPoolSizePerDevice, uint16_t mB
 	// Initialize a DpdkDevice per port
 	for (int i = 0; i < numOfPorts; i++)
 	{
-		DpdkDevice* newDevice = new DpdkDevice(i, mBufPoolSizePerDevice, mBufDataSize);
+		auto newDevice = std::unique_ptr<DpdkDevice>(new DpdkDevice(i, mBufPoolSizePerDevice, mBufDataSize));
 		PCPP_LOG_DEBUG("DpdkDevice #" << i << ": Name='" << newDevice->getDeviceName() << "', PCI-slot='" << newDevice->getPciAddress() << "', PMD='" << newDevice->getPMDName() << "', MAC Addr='" << newDevice->getMacAddress() << "'");
-		m_DpdkDeviceList.push_back(newDevice);
+		m_DeviceList.pushBack(std::move(newDevice));
```

Comment:
Is `std::move` needed? If so, KniDeviceList.cpp:105 also needs it.

Reply:
Move is needed here because `newDevice` is a `unique ptr`. In KniDevice, I explicitly left it as a raw ptr, because the private delete on KniDevice was causing issues with the unique_ptr and it needed to be returned as a result afterwards.

Reply:
sure.

Reply:
Why should this be a `unique_ptr` and not a raw pointer? 🤔

Reply:
I try to use unique ptr whereever possible for the RAII deletion and cleared ownership semantics.